app/views/external_users/admin/external_users/_form.html.haml

```diff
@@ -26,21 +26,28 @@
     - if current_user.persona.admin?
       = f.govuk_check_boxes_fieldset :roles, legend: { text: t('.roles'), size: 's' } do
         - @external_user.available_roles.each_with_index do |role, idx|
-          = f.govuk_check_box(:roles,
-            role,
-            label: { text: role.humanize },
-            link_errors: idx.eql?(0)) do
-
-            - if role.eql?('advocate') && current_user.persona.provider.chamber?
-              = f.govuk_collection_radio_buttons :vat_registered,
-                [['Yes','true'],['No','false']],
-                :last,
-                :first,
-                inline: true,
-                legend: { text: t('.vat_registered'), size: 's' }
-
-    = f.govuk_text_field :supplier_number,
-      hint: { text: t('.supplier_number_hint') },
-      label: { text: t('.supplier_number') }
+          - if role.eql?('advocate')
```

Comment:
i think this should go further down should it not. and remove the else

Reply:
I don't think so, this is how it was before, within the `govuk_check_box` block, thereby creating a 'conditionally revealing' checkbox.
This caused the page height to be resized as we encountered when clicking these checkboxes.
@jsugarman

Reply:
is the supplier number only supposed to be visible under the advocates checkbox?

if so then it looks like this is unnecessarily repeating a block of code in the else. 

Am going to take a look locally

Reply:
I think we need to deal with  page height resizing (or whatever is causing the minor movement issue when checking checkboxes) directly, rather than repeating code statements to work around it. You get that glitch anyway in an agfs firm when selecting the advocate check box on this branch.

Reply:
So the `aria-expanded` of the hidden checkbox input is changed to true. but its the `govuk-checkboxes__conditional--hidden` being removed that causes the "glitch". Is that something we have control over.